server/lib/pbench/test/__init__.py

```diff
@@ -0,0 +1,7 @@
+"""Provides functions, definitions, and PyTest fixtures
+which are common across all Pbench Server unit tests."""
+from pathlib import Path
+
+tarball = "log.tar.xz"
+bad_tarball = "bad_log.tar.xz"
+fixture_path = Path("./server/lib/pbench/test/fixtures/upload")
```

Comment:
You should reevaluate whether you still need this file and these definitions.  (Also, do you need `server/lib/pbench/test/unit/__init__.py`?)

Your code isn't quite consistent about using symbolic accesses vs using literal strings.  However, I'm not sure that there is any need for consistency between test cases, in which case you don't need global definitions for `tarball` and `bad_tarball` (although, they might be convenient to have, if you want to make use of them).  Moreover, if you've successfully mocked out all of the file system accesses (I think there's one that you missed 😉), then you don't need to use any particular value for `fixture_path` -- each of your tests can use whatever path they like.

Now, _within_ a given test, you'll probably want to have paths in variables, so that they can be shared between mocks and so that they can be looked for in messages, etc. -- you want to avoid using repeated string literals -- but there is no requirement to use the same values in every test, so you won't need globals like these.  (These globals were important when the tests relied on artifacts in the file system, but now that you've freed us from that weight, we can afford to jettison these definitions, as well.)

Also, I don't see anything which is going to create the `server/lib/pbench/test/fixtures/upload` directory.  That is, you cannot commit _directories_ to Git -- it creates them as needed to hold committed files, and otherwise it ignores them -- and I don't see any changes to the unit test framework which would create this directory for you.  So, if your code depends on it existing, you need to make the appropriate provisions to ensure that it gets created somehow.  But, my recommendation would be to instead add the necessary mocks so that the test can run without it actually being present in the files system.

Reply:
> you want to avoid using repeated string literals -- but there is no requirement to use the same values in every test, so you won't need globals like these. (These globals were important when the tests relied on artifacts in the file system)

yes, you are right but I find it more convenient to use global values than to define a diff value for each and every test. Also, we won't have to think for a string literal for every TB in mock and define them locally. I prefer it this way.

Yes, I forgot to mock is_dir which was checking for receive_directory with `-002` suffix. I have added a mock for that which relieves us from keeping that directory to pass the check.

Reply:
> I find it more convenient to use global values than to define a diff value for each and every test.

That's fair (although, there is no requirement that each test use a different value from the others, but, since each test is independent of the others, there's no requirement that they use the same value, either).

> Also, we won't have to think for a string literal for every TB in mock

It's OK to use the same choice of name in each test -- since the tests are completely independent of each other (thanks to mocking the file system accesses), that's not actually "repeating yourself".

> we won't have to [...] define them locally. I prefer it this way.

That's fine; however, you should be cognizant of the scope:  unless there is a need to have these variables span test _modules_ they probably shouldn't be defined in the `__init__.py` file (or `conftest.py` or `commons.py`) -- they should just be defined at the top of the module which uses them.  (That is, you should always prefer the smallest or most-restricted scope that will address your needs.)

> I forgot to mock `is_dir` which was checking for `receive_directory` with `-002` suffix. I have added a mock for that which relieves us from keeping that directory to pass the check.

Excellent!  So, make sure that you've deleted the directory from your test environment, since it won't be there in others' test environments...which brings up an awkward point -- the Jenkins testing for this PR is _not_, as far as I can tell, running your new tests.  This branch needs to be based on your #2441 one to enable that.  (I.e., if you rebase this PR's branch on that PR's branch, then its change will be automatically included in this PR which should cause Mr. Jenkins to exercise your new code, and it will allow you to close that PR.)

Reply:
Does this PR need to include the (empty) `server/lib/pbench/test/__init__.py` file?  (Does that drive PyTest or something?)

Reply:
> Does this PR need to include the (empty) server/lib/pbench/test/__init__.py file? (Does that drive PyTest or something?)

Yes, it is important. If I am removing it tests are failing.